File: WorldFeed/Server/src/WorldFeed.Common/DTO/MessageCreateDTO.cs
namespace WorldFeed.Common.DTO
{
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;

    public class MessageCreateDTO : IMessageCreateDTO
    {
        // Twitter fields
        [JsonProperty("target")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IMessageCreateTargetDTO Target { get; set; }

        [JsonProperty("sender_id")]
        public long SenderId { get; set; }

        [JsonProperty("source_app_id")]
        public long? SourceAppId { get; set; }

        [JsonProperty("message_data")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IMessageDataDTO MessageData { get; set; }
    }
}

File: WorldFeed/Client/src/app/core/Core/DTO/MessageCreateDTO.ts
import {IMessageCreateTargetDTO} from "../../Public/Models/Interfaces/DTO/IMessageCreateTargetDTO";
import {IMessageCreateDTO} from "../../Public/Models/Interfaces/DTO/IMessageCreateDTO";
import {IMessageDataDTO} from "../../Public/Models/Interfaces/DTO/IMessageDataDTO";

export class MessageCreateDTO implements IMessageCreateDTO {
  // Twitter fields

  // [JsonProperty("target")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public target: IMessageCreateTargetDTO;

  // [JsonProperty("sender_id")]
  public senderId: number;

  // [JsonProperty("source_app_id")]
  public sourceAppId?: number;

  // [JsonProperty("message_data")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public messageData: IMessageDataDTO;
}
